source/2_platform/rex_opengl/include/rex_opengl/resources/buffer.h

```diff
@@ -16,8 +16,13 @@ namespace rex
     {
     public:
         Buffer(u32 target);
+        Buffer(const Buffer&) = delete;
+        Buffer(Buffer&&) = default;
         ~Buffer() override;
 
+        Buffer& operator=(const Buffer&) = delete;
+        Buffer& operator=(Buffer&&) = default;
```

Comment:
same as for move ctor, but this one should act as if the destructor of *this is called, and then the move ctor is called

Reply:
There should not be any destruction here, there should ony be a move of the resource id
If I call glDeleteBuffer on the given id the buffer that we moved to will also be invalid